index.js

```diff
@@ -140,14 +148,193 @@ app.post('/donate', function(req, res) {
 });
 
 app.post('/giveAgain', function(req, res) {
-  let email = req.body.email;
-  let countrySelection = req.body.countrySelection;
-  let donorCountries = req.body.donorCountries;
+  const email = req.body.email;
+  const countrySelection = req.body.countrySelection;
+  const donorCountries = req.body.donorCountries;
 
   console.log(email, countrySelection, donorCountries);
   res.json({action: 'switch-to-regions'});
 });
 
+app.get('/admin', function(req, res) {
+  if (req.session.loggedin) {
+    res.render('admin');
+  } else {
+    res.render('admin-login');
+  }
+});
+
+app.post('/auth', function(req, res) {
+  const username = req.body.username;
+  const password = req.body.password;
+  if (username && password) {
+    const fileContent = fs.readFileSync('./keys/admin.json');
+    const adminObj = JSON.parse(fileContent);
+    if (adminObj.hasOwnProperty(username) && adminObj[username] === password) {
+      req.session.loggedin = true;
+      req.session.username = username;
+      res.redirect('/admin');
+    } else {
+      res.send('Incorrect Username and/or Password!');
+    }
+    res.end();
+  } else {
+    res.send('Please enter Username and Password');
+    res.end();
+  }
+});
+
+app.post('/getAllCountriesList', function(req, res) {
+  if (!req.session.loggedin) {
+    res.redirect('/admin');
+  }
+  const dbRef = firestore.collection('loc_ref');
+  let countryNames = [];
+  dbRef.get().then((querySnapshot)=>{
+    if (!querySnapshot) {
+      return undefined;
+    }
+    querySnapshot.forEach(function(doc) {
+      countryNames.push(doc.data().country);
+    });
+    res.json({countryNames: countryNames});
+  }).catch((err)=>{
+    console.error(err);
+  });
+});
+
+app.post('/getAllCountryRegions', function(req, res) {
+  if (!req.session.loggedin) {
+    res.redirect('/admin');
+  }
+  const country = req.body.country;
+  const dbRef = firestore.collection('loc_ref').doc(country);
+  let regionData = [];
+  dbRef.get().then((doc)=>{
+    if (!doc.exists) {
+      return undefined;
```

Comment:
Same issue here as on line 195. `app.post` and `app.get` must always return a value to the browser or the window will hang